### DISCUSSION TITLE
Vue warning  - need to add `app.config.unwrapInjectedRef = true`

### DISCUSSION THREAD

**Discussion by CrawfordW** [quasarframework/quasar#11761] [General - Components / Directives / etc] [answered]
I keep getting these Vue warnings:

injected property "iconSetter" is a ref and will be auto-unwrapped and no longer needs `.value` in the next minor release. To opt-in to the new behavior now, set `app.config.unwrapInjectedRef = true`

Can you tell me how/where to set this in Quasar so I can get rid of these warnings? I assume it should go somewhere in the quasar.conf.js file.

**Comment by hawkeye64** [comment]
I'm not see this at all. Have you added Vue to your own deps in package.json?
If you want to get rid of the warning, write a small boot file. When `app` is passed in, then `app.config.unwrapInjectedRef = true`

Reference: https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/src/componentOptions.ts#L853-L857

**Reply by CrawfordW** [reply]
Haven't added Vue in package.json. Should I?
I tried adding a boot file to suppress those warnings, but now I get:
Unhandled error during execution of render function 
Maybe I'll just have to live with those annoying warnings - at least everything works.

**Comment by MilosPaunovic** [comment]
Which specific version of `Vue` is installed in your environment?

The warning message you're seeing is related to this file change:
[https://github.com/vuejs/vue-next/blob/741d3b36f21582e682009114961b5cd3146e8dad/packages/runtime-core/src/componentOptions.ts#L853](https://github.com/vuejs/vue-next/blob/741d3b36f21582e682009114961b5cd3146e8dad/packages/runtime-core/src/componentOptions.ts#L853)

And was a part of [3.2.0-beta-6 release ](https://github.com/vuejs/vue-next/blob/741d3b36f21582e682009114961b5cd3146e8dad/packages/runtime-core/src/componentOptions.ts#L853)

**Reply by CrawfordW** [reply]
I have the latest Quasar release including Vue 3.2.26.
The warning says it's only until the next minor release, but I've been seeing it for months!
The way I'm using this is to pass 'iconSetter' as a prop down through several layers using provide like this:
```
props: {
   iconSetter: {
      type: Function,
      default: null
   }
},
provide () {
   return {
      iconSetter: computed(() => this.iconSetter)
   }
}
```
Perhaps I'm doing something wrong?

**Reply by MilosPaunovic** [reply]
Next minor release would be `Vue@3.3.x`, so you should check this again after that is released and updated as a part of `@quasar/app` dependencies.

**Reply by CrawfordW** [reply]
OK. Thanks for the input.

**Reply by CrawfordW** [reply]
Decided to drop the use of 'provided' feature and just pass the prop down through multiple levels of components. No longer getting those annoying warnings.

**Comment by Jille** [comment]
I created this `src/boot/vueconfig.ts` and referenced it from quasar.conf.js

```
import { boot } from 'quasar/wrappers'

export default boot(({ app }) => {
  app.config.unwrapInjectedRef = true;
})
```

Seems to work.

**Comment by rstoenescu** [comment, marked as answer]
Vue should be a direct dependency of your app. It is NOT supplied by Quasar CLI. If there is a problem with vue it should be reported to Vue Team.
